### ISSUE TITLE
Support for icc 17

### ISSUE THREAD

**Issue by rakhimov** [nemequ/icc-travis#10]
The version of icc is hardcoded in the script.
icc 17 is available and updates will be coming.

General solution would be to install the latest version,
but how to know what the latest is (without hardcoding)?

**Comment by rakhimov** [comment]
Maybe some env var like ``INTEL_COMPILER_VERSION``, ``INTEL_COMPILER_UPDATE``.
This is still not perfect, though.

**Comment by nemequ** [comment]
When 17 is stable, it should be the default.  16 could be selectable with a command line argument (--version would probably be okay).  All the individual package names need to be changed, so I'll have to go through them all again and try to figure out what goes with what component.

**Comment by rakhimov** [comment]
@nemequ That sounds tedious.

**Comment by rakhimov** [comment]
I got some configurations for the icc only: https://github.com/rakhimov/icc-travis/tree/icc17
17 version has some component naming conflicts.

I think it is possible to get components from mediaconfig without hardcoding.
``grep 'Abbr' mediconfig | sed 'strip tags' | grep 'icc-pattern'``,
then do some joining to get the final list.

**Comment by rakhimov** [comment]
17.1 is out.

**Comment by jeffhammond** [comment]
17.0.[1234] are all released, as is Intel 18 beta.  I'll try to capture all the paths and such.

**Comment by jeffhammond** [comment]
@rakhimov Can you submit a pull request to fire up Travis testing for your changes?

**Comment by nemequ** [comment]
Does 18 remove the registration when you uninstall it?  This script doesn't really work for me anymore since Intel now [limits activations](https://github.com/nemequ/icc-travis/issues/3#issuecomment-258502782) to (IIRC) 5 machines.  Without a way to deactivate a license automatically (uninstallation doesn't do the trick), you have to log into Intel's web site and delete activations manually every 5 builds.

Basically, I'm hesitant to spend any time upgrading to 17 or 18 unless there is a way to automate everything again.  That said, I'm quite willing to accept patches.

**Comment by jeffhammond** [comment]
I wasn't aware of this issue.  I will inquire internally.

**Comment by rakhimov** [comment]
Besides the licensing problem,
I find it tedious to upgrade from one version to another manually
by going through all the component changes.
I hope someone comes up with a smart script that would pick the components automatically.

**Comment by jeffhammond** [comment]
@rakhimov I guess the easy button for that issue would be to just install everything.  One can optimize this by downloading Composer instead of e.g. Parallel Studio Cluster Edition, because Composer does not include the performance analysis tools that aren't likely to be used in Travis.  (https://software.intel.com/en-us/intel-parallel-studio-xe has details)

I can't comment on the details, but there is an active effort to make all of this easier.

**Comment by nemequ** [comment]
Closing since this issue is obsolete now that the APT repository works.
